packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -128,6 +137,12 @@ export default async function initSanity(
     throw new Error('`--reconfigure` is deprecated - manual configuration is now required')
   }
 
+  // use vercel's framework detector
+  const detectedFramework: Framework | null = await detectFrameworkRecord({
+    fs: new LocalFileSystemDetector(workDir),
+    frameworkList: frameworks as readonly Framework[],
+  })
+
```

Comment:
Moved this higher up so we have access to it earlier. Otherwise, we couldn't have detected Next.js.